test/compilable/test22626.d

```diff
@@ -0,0 +1,11 @@
+// https://issues.dlang.org/show_bug.cgi?id=22626
+// REQUIRED_ARGS: -preview=nosharedaccess
+
+class Oops
+{
+    shared int a;
+    synchronized void oops()
+    {
+        a = 2;
```

Comment:
please make this 
```d
shared int* pa;
shared int a;
synchronized void oops()
{
    a = 2; // OK
    static assert (!__traits(compiles,*pa = 2));
}
```
to ensure that only one level of dereferencing works.

Also please add back in the global variable case as also failing to compile.